diff --git a/host/utils/CMakeLists.txt b/host/utils/CMakeLists.txt
index 4bbcd252b..03536e12c 100644
--- a/host/utils/CMakeLists.txt
+++ b/host/utils/CMakeLists.txt
@@ -21,7 +21,7 @@ set(util_runtime_sources
 #for each source: build an executable and install
 foreach(util_source ${util_runtime_sources})
     get_filename_component(util_name ${util_source} NAME_WE)
-    add_executable(${util_name} ${util_source})
+    add_library(${util_name} ${util_source})
     target_link_libraries(${util_name} uhd ${Boost_LIBRARIES})
     UHD_INSTALL(TARGETS ${util_name} RUNTIME DESTINATION ${RUNTIME_DIR} COMPONENT utilities)
 endforeach(util_source)
@@ -87,7 +87,7 @@ endif(LINUX AND ENABLE_USB)
 #for each source: build an executable and install
 foreach(util_source ${util_share_sources})
     get_filename_component(util_name ${util_source} NAME_WE)
-    add_executable(${util_name} ${util_source})
+    add_library(${util_name} ${util_source})
     target_link_libraries(${util_name} uhd ${Boost_LIBRARIES})
     UHD_INSTALL(TARGETS ${util_name} RUNTIME DESTINATION ${PKG_LIB_DIR}/utils COMPONENT utilities)
 endforeach(util_source)
